play-ahc-ws-standalone/src/main/scala/play/api/libs/ws/ahc/StandaloneAhcWSRequest.scala

```diff
@@ -66,21 +70,20 @@ case class StandaloneAhcWSRequest(
     copy(auth = Some((username, password, scheme)))
 
   override def withHttpHeaders(hdrs: (String, String)*): Self = {
-    var newHeaders = hdrs.foldLeft(TreeMap[String, Seq[String]]()(CaseInsensitiveOrdered)) {
+    val newHeaders = hdrs.foldLeft(TreeMap[String, Seq[String]]()(CaseInsensitiveOrdered)) {
       (m, hdr) =>
         if (m.contains(hdr._1)) m.updated(hdr._1, m(hdr._1) :+ hdr._2)
         else m + (hdr._1 -> Seq(hdr._2))
     }
 
-    // preserve the content type
-    newHeaders = contentType match {
-      case Some(ct) =>
-        newHeaders.updated(HttpHeaders.Names.CONTENT_TYPE, Seq(ct))
+    // preserve the existing content type or override if the new headers contains a new one.
```

Comment:
Hmmm, if `withHeaders` removes any previously existing headers I think I'm missing a method to `append`/`add` new header pairs to the map.